gui/src/lib.rs

```diff
@@ -189,11 +192,11 @@ pub fn is_string_numeric(word: &str) -> bool {
     return true;
 }
 
-pub fn get_file_name_from_path(file_path: &str) -> &str {
+pub fn get_file_name_from_path(file_path: &str) -> String {
     let location_paths: Vec<&str> = file_path.split(path::MAIN_SEPARATOR).collect();
     let file_name = location_paths[location_paths.len() - 1];
 
-    file_name
+    file_name.to_string()
 }
```

Comment:
btw, why you're returning a `String` instead of `&str` here and constructing a string from the string literal.

I see that we don't have any use cases for `String` type and we always need the return value from `get_file_name_from_path` to be `&str` in every place. Also constructing a string is slower than taking a partial value from the passed string literal, so It's also better from a performance perspective.

Reply:
I ran into a lifetime issue with the string reference and I was kind of whirlwind fixing things at the moment. I'll look at it again later, and try to see if it was actually necessary.